src/librbd/ImageCtx.cc

```diff
@@ -849,8 +849,11 @@ struct C_InvalidateCache : public Context {
     // ensure no locks are held when flush is complete
     on_safe = util::create_async_context_callback(*this, on_safe);
 
+    bool journaling = false;
+    journaling = (journal != NULL && !journal->is_journal_replaying());
+
```

Comment:
Isn't it better to set this below the assert? I guess journal is safe to access only under the lock?